lib/ui/navbar.js

```diff
@@ -245,60 +217,62 @@ export default class IpNavBar extends React.Component {
   }
 
   connectOrDisconnect() {
-    this.doConnectOrDisconnect(this.state.host);
-  }
-
-  doConnectOrDisconnect(host) {
-    if (host) {
-      if (!(this.state.connecting || this.state.connected)) {
-        const useSSL = host.startsWith('https://') || window.location.protocol === 'https:'
-        const theHost = host.startsWith('https:')  || host.startsWith('http:') ? host.split(':').slice(1).join(':').replace(/\//g, '') : host
-        SignalkClient = new Client(host, 3000, useSSL) //only useSSL is relevant, host & port overridden below
-        this.props.instrumentPanel.useSSL = useSSL;
-        var thisOnConnect = this.onConnect;
-        this.setState({
+    var that = this;
+    const SignalkClient = this.props.instrumentPanel.SignalkClient;
+    if (!(this.state.connecting || this.state.connected)) {
+      SignalkClient.connect();
+      this.setState({
+        "connecting": true,
+      });
+      SignalkClient.on('connect', () => {
+        SignalkClient.subscribe({context:'vessels.self',subscribe:[{ path:'*'}]});
+        SignalkClient.on('delta', this.props.instrumentPanel.dispatch.bind(this.props.instrumentPanel));
+        that.onConnect();
+      });
+      SignalkClient.on('error', (err) => {
+        console.log(err);
+        that.setState({
           "connecting": true,
-          "host": host,
-          "connection": SignalkClient.connectDelta(theHost,
-            this.props.instrumentPanel.dispatch.bind(this.props.instrumentPanel),
-            function(skConnection) {
-              skConnection.subscribeAll();
-              thisOnConnect(host);
-            },
-            this.onDisconnect,
-            function(error) {
-              console.log(error)
-            },
-            'self'
-          ),
-          "hosts": this.addToHosts(host, this.state.hosts)
+          "connected": false
         });
-        try {
-          window.localStorage['signalKLastHost'] = host;
-          window.localStorage['signalKHostConnected'] = true;
-        } catch (ex) {
-          console.error(ex)
-        }
-      } else {
+      });
+      SignalkClient.on('hitMaxRetries', () => {
+        console.log('hitMaxRetries');
+        that.setState({
+          "connecting": false,
+          "connected": false
+        });
+        alert("Cannot connect to signal k server: ws" +
```

Comment:
Signal K is spelled _Signal K_, could you please fix all messages to use it properly.

Reply:
fixed by commit: 56f344f